portfolio/src/main/java/com/google/sps/servlets/PokedexDataServlet.java

```diff
@@ -10,27 +13,38 @@
 
 @WebServlet("/pokedex-data")
 public class PokedexDataServlet extends HttpServlet {
-  private static final int NUMBER_OF_GENERATIONS =
-      7; // 6 generations, Array Index directly  (1st Gen -> index of 1)
+  private static final int NUMBER_OF_POKEMON = 800;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json");
-    int[] generations = getGenerationCount();
+    HashMap<String, ArrayList<Integer>> stats = getStats();
     Gson gson = new Gson();
-    String json = gson.toJson(generations);
+    String json = gson.toJson(stats);
     response.getWriter().println(json);
   }
 
-  private int[] getGenerationCount() {
-    int[] result = new int[NUMBER_OF_GENERATIONS];
+  private HashMap<String, ArrayList<Integer>> getStats() {
+    HashMap<String, ArrayList<Integer>> result = new HashMap<String, ArrayList<Integer>>();
+    String[] fields = {"HP", "Attack", "Defense", "Sp. Atk", "Sp. Def", "Speed"};
+
+    for (String field : fields) {
+      result.put(field, new ArrayList<Integer>(Collections.nCopies(NUMBER_OF_POKEMON, 0)));
+    }
+
     Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/pokemon.csv"));
     String header = scanner.nextLine();
-    int generationColIndex = getColIndex(header, "Generation");
+    int entryNumColIndex = getColIndex(header, "#");
+
     while (scanner.hasNextLine()) {
       String line = scanner.nextLine();
       String[] cells = line.split(",");
-      result[Integer.valueOf(cells[generationColIndex])]++;
+      int pokedexEntryNum = Integer.valueOf(cells[entryNumColIndex]);
```

Comment:
this took a lot of brain RAM to understand the flow here.

some explanatory comments in the logic would go a long way here.

imagine returning to this code in a few months and trying to understand.

I think the most important thing to document, is your result object, what the values in the array list are representing.